app/Actions/Album/Tracks.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace App\Actions\Album;
+
+use App\Models\Album;
+use Illuminate\Database\Eloquent\ModelNotFoundException;
+use Illuminate\Http\UploadedFile;
+use Illuminate\Support\Facades\Storage;
+
+class Tracks extends Action
+{
+	/**
+	 * @param string $albumID the ID of the album
+	 * @param mixed  $value   the value to be set
+	 */
+	public function set(string $albumID, UploadedFile $value): void
+	{
+		$album = $this->albumFactory->findAlbumOrFail($albumID, false);
```

Comment:
Please don't pass around the ID of the album as string. The new approach is to fetch the required album as part of the request, so we can properly fail with an HTTP error 422, 404, 401 or 403 in the correct order (see classes below `\App\Http\Requests`). This way the instance of the album model becomes part of the request and we only need to query the DB once.